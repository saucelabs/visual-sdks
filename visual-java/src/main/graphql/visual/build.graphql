query build($input: UUID!) {
  result: build(id: $input) {
    url
  }
}
