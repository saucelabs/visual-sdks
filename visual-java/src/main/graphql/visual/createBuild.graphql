mutation createBuild($input: BuildIn!) {
  result: createBuild(input: $input) {
    id
    name
    project
    branch
    status
    url
  }
}
