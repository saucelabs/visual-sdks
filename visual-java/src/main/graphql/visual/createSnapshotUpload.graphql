mutation createSnapshotUpload($input: SnapshotUploadIn!) {
  result: createSnapshotUpload(input: $input) {
    id
    buildId
    imageUploadUrl
  }
}
