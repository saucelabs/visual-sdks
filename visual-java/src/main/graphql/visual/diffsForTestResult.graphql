query diffsForTestResult($input: UUID!) {
  result: diffs(condition: { buildId: $input }) {
    nodes {
      id
      status
    }
  }
}
