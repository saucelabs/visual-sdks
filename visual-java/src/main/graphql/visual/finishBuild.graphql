mutation FinishBuild($input: FinishBuildIn!) {
  result: finishBuild(input: $input) {
    id
    name
    status
    url
  }
}
