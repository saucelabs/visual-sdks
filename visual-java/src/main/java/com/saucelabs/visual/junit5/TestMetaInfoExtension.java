package com.saucelabs.visual.junit5;

import com.saucelabs.visual.TestMetaInfo;
import java.util.Optional;
import org.junit.jupiter.api.extension.AfterTestExecutionCallback;
import org.junit.jupiter.api.extension.BeforeTestExecutionCallback;
import org.junit.jupiter.api.extension.ExtensionContext;

public class TestMetaInfoExtension
    implements BeforeTestExecutionCallback, AfterTestExecutionCallback {
  @Override
  public void beforeTestExecution(ExtensionContext context) throws Exception {
    String className = context.getRequiredTestClass().getName();
    String testName = context.getRequiredTestMethod().getName();

    TestMetaInfo.THREAD_LOCAL.set(Optional.of(new TestMetaInfo(className, testName)));
  }

  @Override
  public void afterTestExecution(ExtensionContext context) throws Exception {
    TestMetaInfo.THREAD_LOCAL.set(Optional.empty());
  }
}
