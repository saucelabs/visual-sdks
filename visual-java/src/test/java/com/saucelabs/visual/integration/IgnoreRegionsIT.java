package com.saucelabs.visual.integration;

import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.junit5.SnapshotExtension;
import com.saucelabs.visual.CheckOptions;
import com.saucelabs.visual.junit5.TestMetaInfoExtension;
import com.saucelabs.visual.model.FullPageScreenshotConfig;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.openqa.selenium.By;

@ExtendWith({TestMetaInfoExtension.class, SnapshotExtension.class})
public class IgnoreRegionsIT extends IntegrationBase {
  Expect expect;

  @BeforeAll
  public static void login() {
    LoginPage loginPage = new LoginPage();
    loginPage.open();
    loginPage.login();
    InventoryLongPage inventoryPage = new InventoryLongPage();
    inventoryPage.open();
  }

  final By ignoreSelectors = By.cssSelector(".inventory_item_img,.btn_inventory");

  @Test
  public void checkDefaultStandardIgnoreRegions() {
    String id =
        sauceVisualCheck(
            "Standard",
            new CheckOptions.Builder()
                .withIgnoreElements(driver.findElements(ignoreSelectors))
                .build());
    String result = getSnapshotResult(id);
    expect.toMatchSnapshot(result);
  }

  @Test
  public void checkPositionRegionsOnClippedSnapshot() {
    String id =
        sauceVisualCheck(
            "Clipped",
            new CheckOptions.Builder()
                .withClipElement(driver.findElement(By.cssSelector(".inventory_list")))
                .withIgnoreElements(driver.findElements(ignoreSelectors))
                .build());
    String result = getSnapshotResult(id);
    expect.toMatchSnapshot(result);
  }

  @Test
  public void checkPositionRegionsOnFPS() {
    String id =
        sauceVisualCheck(
            "FPS",
            new CheckOptions.Builder()
                .withFullPageConfig(new FullPageScreenshotConfig.Builder().build())
                .withIgnoreElements(driver.findElements(ignoreSelectors))
                .build());
    String result = getSnapshotResult(id);
    expect.toMatchSnapshot(result);
  }

  @Test
  public void checkPositionRegionsOnFPSWhenClipped() {
    String id =
        sauceVisualCheck(
            "Clipped (FPS)",
            new CheckOptions.Builder()
                .withFullPageConfig(new FullPageScreenshotConfig.Builder().build())
                .withIgnoreElements(driver.findElements(ignoreSelectors))
                .withClipElement(driver.findElement(By.cssSelector(".inventory_list")))
                .build());
    String result = getSnapshotResult(id);
    expect.toMatchSnapshot(result);
  }

  @Test
  public void checkPositionRegionsOnFPSWhenScrolledPriorToClip() {
    driver.executeScript("window.scrollBy(0, 100)");

    String id =
        sauceVisualCheck(
            "FPS (scrolled)",
            new CheckOptions.Builder()
                .withFullPageConfig(new FullPageScreenshotConfig.Builder().build())
                .withIgnoreElements(driver.findElements(ignoreSelectors))
                .withClipElement(driver.findElement(By.cssSelector(".inventory_list")))
                .build());
    String result = getSnapshotResult(id);
    expect.toMatchSnapshot(result);
  }
}
