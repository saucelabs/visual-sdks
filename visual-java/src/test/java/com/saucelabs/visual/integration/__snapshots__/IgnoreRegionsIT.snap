com.saucelabs.visual.integration.IgnoreRegionsIT.checkDefaultStandardIgnoreRegions=[
{
  "snapshot" : {
    "name" : "Standard (Sauce)",
    "suiteName" : "com.saucelabs.visual.integration.IgnoreRegionsIT",
    "testName" : "checkDefaultStandardIgnoreRegions",
    "operatingSystem" : "WINDOWS",
    "operatingSystemVersion" : "10",
    "device" : "Desktop (1600x1059)",
    "devicePixelRatio" : 1,
    "hasDom" : true,
    "diffs" : {
      "nodes" : [ {
        "status" : "EQUAL",
        "diffClusters" : [ ]
      } ]
    },
    "ignoreRegions" : [ {
      "x" : 271,
      "y" : 155,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 271,
      "y" : 155,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 580,
      "y" : 339,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 826,
      "y" : 155,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 826,
      "y" : 155,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 1135,
      "y" : 339,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 271,
      "y" : 407,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 271,
      "y" : 407,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 580,
      "y" : 591,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 826,
      "y" : 407,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 826,
      "y" : 407,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 1135,
      "y" : 591,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 271,
      "y" : 659,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 271,
      "y" : 659,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 580,
      "y" : 843,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 826,
      "y" : 659,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 826,
      "y" : 659,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 1135,
      "y" : 843,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 271,
      "y" : 911,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 271,
      "y" : 911,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 826,
      "y" : 911,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 826,
      "y" : 911,
      "width" : 159,
      "height" : 238
    } ]
  }
}
]


com.saucelabs.visual.integration.IgnoreRegionsIT.checkPositionRegionsOnClippedSnapshot=[
{
  "snapshot" : {
    "name" : "Clipped (Sauce)",
    "suiteName" : "com.saucelabs.visual.integration.IgnoreRegionsIT",
    "testName" : "checkPositionRegionsOnClippedSnapshot",
    "operatingSystem" : "WINDOWS",
    "operatingSystemVersion" : "10",
    "device" : "Desktop (1600x1059)",
    "devicePixelRatio" : 1,
    "hasDom" : true,
    "diffs" : {
      "nodes" : [ {
        "status" : "EQUAL",
        "diffClusters" : [ ]
      } ]
    },
    "ignoreRegions" : [ {
      "x" : 16,
      "y" : 1,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 16,
      "y" : 1,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 325,
      "y" : 185,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 571,
      "y" : 1,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 571,
      "y" : 1,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 880,
      "y" : 185,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 16,
      "y" : 253,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 16,
      "y" : 253,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 325,
      "y" : 437,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 571,
      "y" : 253,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 571,
      "y" : 253,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 880,
      "y" : 437,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 16,
      "y" : 505,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 16,
      "y" : 505,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 325,
      "y" : 689,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 571,
      "y" : 505,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 571,
      "y" : 505,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 880,
      "y" : 689,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 16,
      "y" : 757,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 16,
      "y" : 757,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 325,
      "y" : 941,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 571,
      "y" : 757,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 571,
      "y" : 757,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 880,
      "y" : 941,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 16,
      "y" : 1009,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 16,
      "y" : 1009,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 571,
      "y" : 1009,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 571,
      "y" : 1009,
      "width" : 159,
      "height" : 238
    } ]
  }
}
]


com.saucelabs.visual.integration.IgnoreRegionsIT.checkPositionRegionsOnFPS=[
{
  "snapshot" : {
    "name" : "FPS (Sauce)",
    "suiteName" : "com.saucelabs.visual.integration.IgnoreRegionsIT",
    "testName" : "checkPositionRegionsOnFPS",
    "operatingSystem" : "WINDOWS",
    "operatingSystemVersion" : "10",
    "device" : "Desktop (1600x1059)",
    "devicePixelRatio" : 1,
    "hasDom" : true,
    "diffs" : {
      "nodes" : [ {
        "status" : "EQUAL",
        "diffClusters" : [ ]
      } ]
    },
    "ignoreRegions" : [ {
      "x" : 271,
      "y" : 155,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 271,
      "y" : 155,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 580,
      "y" : 339,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 826,
      "y" : 155,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 826,
      "y" : 155,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 1135,
      "y" : 339,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 271,
      "y" : 407,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 271,
      "y" : 407,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 580,
      "y" : 591,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 826,
      "y" : 407,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 826,
      "y" : 407,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 1135,
      "y" : 591,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 271,
      "y" : 659,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 271,
      "y" : 659,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 580,
      "y" : 843,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 826,
      "y" : 659,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 826,
      "y" : 659,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 1135,
      "y" : 843,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 271,
      "y" : 911,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 271,
      "y" : 911,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 580,
      "y" : 1095,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 826,
      "y" : 911,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 826,
      "y" : 911,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 1135,
      "y" : 1095,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 271,
      "y" : 1163,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 271,
      "y" : 1163,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 580,
      "y" : 1347,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 826,
      "y" : 1163,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 826,
      "y" : 1163,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 1135,
      "y" : 1347,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 271,
      "y" : 1415,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 271,
      "y" : 1415,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 580,
      "y" : 1599,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 826,
      "y" : 1415,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 826,
      "y" : 1415,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 1135,
      "y" : 1599,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 271,
      "y" : 1667,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 271,
      "y" : 1667,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 580,
      "y" : 1851,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 826,
      "y" : 1667,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 826,
      "y" : 1667,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 1135,
      "y" : 1851,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 271,
      "y" : 1919,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 271,
      "y" : 1919,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 580,
      "y" : 2103,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 826,
      "y" : 1919,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 826,
      "y" : 1919,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 1135,
      "y" : 2103,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 271,
      "y" : 2171,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 271,
      "y" : 2171,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 580,
      "y" : 2355,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 826,
      "y" : 2171,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 826,
      "y" : 2171,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 1135,
      "y" : 2355,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 271,
      "y" : 2423,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 271,
      "y" : 2423,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 580,
      "y" : 2607,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 826,
      "y" : 2423,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 826,
      "y" : 2423,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 1135,
      "y" : 2607,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 271,
      "y" : 2675,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 271,
      "y" : 2675,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 580,
      "y" : 2859,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 826,
      "y" : 2675,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 826,
      "y" : 2675,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 1135,
      "y" : 2859,
      "width" : 160,
      "height" : 34
    } ]
  }
}
]


com.saucelabs.visual.integration.IgnoreRegionsIT.checkPositionRegionsOnFPSWhenClipped=[
{
  "snapshot" : {
    "name" : "Clipped (FPS) (Sauce)",
    "suiteName" : "com.saucelabs.visual.integration.IgnoreRegionsIT",
    "testName" : "checkPositionRegionsOnFPSWhenClipped",
    "operatingSystem" : "WINDOWS",
    "operatingSystemVersion" : "10",
    "device" : "Desktop (1600x1059)",
    "devicePixelRatio" : 1,
    "hasDom" : true,
    "diffs" : {
      "nodes" : [ {
        "status" : "EQUAL",
        "diffClusters" : [ ]
      } ]
    },
    "ignoreRegions" : [ {
      "x" : 16,
      "y" : 1,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 16,
      "y" : 1,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 325,
      "y" : 185,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 571,
      "y" : 1,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 571,
      "y" : 1,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 880,
      "y" : 185,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 16,
      "y" : 253,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 16,
      "y" : 253,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 325,
      "y" : 437,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 571,
      "y" : 253,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 571,
      "y" : 253,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 880,
      "y" : 437,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 16,
      "y" : 505,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 16,
      "y" : 505,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 325,
      "y" : 689,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 571,
      "y" : 505,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 571,
      "y" : 505,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 880,
      "y" : 689,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 16,
      "y" : 757,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 16,
      "y" : 757,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 325,
      "y" : 941,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 571,
      "y" : 757,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 571,
      "y" : 757,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 880,
      "y" : 941,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 16,
      "y" : 1009,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 16,
      "y" : 1009,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 325,
      "y" : 1193,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 571,
      "y" : 1009,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 571,
      "y" : 1009,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 880,
      "y" : 1193,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 16,
      "y" : 1261,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 16,
      "y" : 1261,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 325,
      "y" : 1445,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 571,
      "y" : 1261,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 571,
      "y" : 1261,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 880,
      "y" : 1445,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 16,
      "y" : 1513,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 16,
      "y" : 1513,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 325,
      "y" : 1697,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 571,
      "y" : 1513,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 571,
      "y" : 1513,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 880,
      "y" : 1697,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 16,
      "y" : 1765,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 16,
      "y" : 1765,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 325,
      "y" : 1949,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 571,
      "y" : 1765,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 571,
      "y" : 1765,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 880,
      "y" : 1949,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 16,
      "y" : 2017,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 16,
      "y" : 2017,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 325,
      "y" : 2201,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 571,
      "y" : 2017,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 571,
      "y" : 2017,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 880,
      "y" : 2201,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 16,
      "y" : 2269,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 16,
      "y" : 2269,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 325,
      "y" : 2453,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 571,
      "y" : 2269,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 571,
      "y" : 2269,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 880,
      "y" : 2453,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 16,
      "y" : 2521,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 16,
      "y" : 2521,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 325,
      "y" : 2705,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 571,
      "y" : 2521,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 571,
      "y" : 2521,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 880,
      "y" : 2705,
      "width" : 160,
      "height" : 34
    } ]
  }
}
]


com.saucelabs.visual.integration.IgnoreRegionsIT.checkPositionRegionsOnFPSWhenScrolledPriorToClip=[
{
  "snapshot" : {
    "name" : "FPS (scrolled) (Sauce)",
    "suiteName" : "com.saucelabs.visual.integration.IgnoreRegionsIT",
    "testName" : "checkPositionRegionsOnFPSWhenScrolledPriorToClip",
    "operatingSystem" : "WINDOWS",
    "operatingSystemVersion" : "10",
    "device" : "Desktop (1600x1059)",
    "devicePixelRatio" : 1,
    "hasDom" : true,
    "diffs" : {
      "nodes" : [ {
        "status" : "EQUAL",
        "diffClusters" : [ ]
      } ]
    },
    "ignoreRegions" : [ {
      "x" : 16,
      "y" : 1,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 16,
      "y" : 1,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 325,
      "y" : 185,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 571,
      "y" : 1,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 571,
      "y" : 1,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 880,
      "y" : 185,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 16,
      "y" : 253,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 16,
      "y" : 253,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 325,
      "y" : 437,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 571,
      "y" : 253,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 571,
      "y" : 253,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 880,
      "y" : 437,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 16,
      "y" : 505,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 16,
      "y" : 505,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 325,
      "y" : 689,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 571,
      "y" : 505,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 571,
      "y" : 505,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 880,
      "y" : 689,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 16,
      "y" : 757,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 16,
      "y" : 757,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 325,
      "y" : 941,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 571,
      "y" : 757,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 571,
      "y" : 757,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 880,
      "y" : 941,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 16,
      "y" : 1009,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 16,
      "y" : 1009,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 325,
      "y" : 1193,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 571,
      "y" : 1009,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 571,
      "y" : 1009,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 880,
      "y" : 1193,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 16,
      "y" : 1261,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 16,
      "y" : 1261,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 325,
      "y" : 1445,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 571,
      "y" : 1261,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 571,
      "y" : 1261,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 880,
      "y" : 1445,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 16,
      "y" : 1513,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 16,
      "y" : 1513,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 325,
      "y" : 1697,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 571,
      "y" : 1513,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 571,
      "y" : 1513,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 880,
      "y" : 1697,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 16,
      "y" : 1765,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 16,
      "y" : 1765,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 325,
      "y" : 1949,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 571,
      "y" : 1765,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 571,
      "y" : 1765,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 880,
      "y" : 1949,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 16,
      "y" : 2017,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 16,
      "y" : 2017,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 325,
      "y" : 2201,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 571,
      "y" : 2017,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 571,
      "y" : 2017,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 880,
      "y" : 2201,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 16,
      "y" : 2269,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 16,
      "y" : 2269,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 325,
      "y" : 2453,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 571,
      "y" : 2269,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 571,
      "y" : 2269,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 880,
      "y" : 2453,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 16,
      "y" : 2521,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 16,
      "y" : 2521,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 325,
      "y" : 2705,
      "width" : 160,
      "height" : 34
    }, {
      "x" : 571,
      "y" : 2521,
      "width" : 150,
      "height" : 238
    }, {
      "x" : 571,
      "y" : 2521,
      "width" : 159,
      "height" : 238
    }, {
      "x" : 880,
      "y" : 2705,
      "width" : 160,
      "height" : 34
    } ]
  }
}
]